cloudstack/resource_cloudstack_instance.go

```diff
@@ -277,6 +283,11 @@ func resourceCloudStackInstanceCreate(d *schema.ResourceData, meta interface{})
 		p.SetRootdisksize(int64(rootdisksize.(int)))
 	}
 
+	if d.Get("uefi").(bool) {
+		p.SetBoottype("UEFI")
+		p.SetBootmode("Legacy")
```

Comment:
Should the boot mode be configurable?

Reply:
I was thinking about adding this as a second step, if the community needs it. Currently I don't have any way to test the UEFI Secure mode (I'm not even sure it works properly in ACS). 

For now this is meant to be an easy and naive implementation that covers the most important need, which is to run UEFI cloud images; This is also a pressing issue now that some distros (e.g. Ubuntu) are providing [UEFI only cloud images](https://cloud-images.ubuntu.com/noble/current/).

Reply:
@rohityadavcloud Is this fine?